ang/civicase/CaseListDirectives.js

```diff
@@ -31,9 +31,32 @@
       var $header = $el.find('thead');
 
       (function init () {
+        initWatchers();
+        initSubscribers();
+      }());
+
+      /**
+       * Initialise all watchers
+       */
+      function initWatchers () {
         scope.$watch('isLoading', fixPositioningOnLoadingOrFocusChange);
         scope.$watch('caseIsFocused', fixPositioningOnLoadingOrFocusChange);
-      }());
+      }
+
+      /**
+       * Initialise all subscribers
+       */
+      function initSubscribers () {
+        $rootScope.$on('stickyCaseListHeaderReinitiatePos', function () {
+          $timeout(function () {
```

Comment:
Why do we need `timeout` here?

Reply:
We have to call the logic when DOM has updated and the best way to ensure that is to call the logic after the angular is done with its DOM updates. To make sure it is called after that we need to add a timeout. 

Please let me know if we need to discuss it further.

Reply:
Lets discuss this.